signature/policy_eval.go

```diff
@@ -253,6 +253,32 @@ func (pc *PolicyContext) GetSignaturesWithAcceptedAuthor(ctx context.Context, pu
 	return res, nil
 }
 
+// IsImagePolicySigned true iff the policy requirement for the image is not insecureAcceptAnything
+func (pc *PolicyContext) IsImagePolicySigned(publicImage types.UnparsedImage) (res bool, finalErr error) {
```

Comment:
I'd say let's not replicate the "bool and error" that we now think was a mistake.

IOW, let's call this `RequireSignedPolicyFor` and have it just return `error`.